### PR TITLE
Feature/dacol engtool

### PR DESCRIPTION
**Pull request by dacolgit**
add read/write/call
start integration of publishing nodes

### CODE REVIEW

File: samples/src/Microsoft.Azure.IIoT.App/src/Models/PublishedNode.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+namespace Microsoft.Azure.IIoT.App.Services {
+    using System.Collections.Generic;
+    public class PublishedNode {
+        public string NodeId { get; set; }
+        public int? SampligInterval { get; set; }
```
Comment (mregen) [inline samples/src/Microsoft.Azure.IIoT.App/src/Models/PublishedNode.cs]: >SampligInte [](start = 20, length = 11)

typo

File: samples/src/Microsoft.Azure.IIoT.App/src/Pages/Browser.razor [hunk 2]
```diff
@@ -58,12 +51,12 @@
...
     </thead>
     <tbody>
-        @foreach (var node in pagedNodeList.Results)
+        @foreach (var node in PagedNodeList.Results)
         {
             <tr>
                 <td>@node.NodeName</td>
                 <td>@node.Id</td>
-                @if (node.Children == true)
+                @if ((node.Children == true) && (node.NodeClass != NodeClass.Method))
```
Comment (mregen) [inline samples/src/Microsoft.Azure.IIoT.App/src/Pages/Browser.razor]: >(node.Children == true) [](start = 21, length = 23)

do you really need to do it comparing a bool with a bool. Is Children a bool?

File: samples/src/Microsoft.Azure.IIoT.App/src/Pages/Browser.razor [hunk 3]
```diff
@@ -175,93 +128,157 @@
...
+        var publishedNodes = await Publisher.PublishedAsync(endpointId);
 
-    private void OpenDrawer(ListNode node)
-    {
-        IsOpened = true;
-        nodeData = node;
-    }
+        foreach (var node in NodeList.Results)
+        {
+            // fetch the actual value
```
Comment (mregen) [inline samples/src/Microsoft.Azure.IIoT.App/src/Pages/Browser.razor]: >// fetch the actual value [](start = 12, length = 25)

this comment does not describe the loop action here?

File: samples/src/Microsoft.Azure.IIoT.App/src/Pages/Browser.razor [hunk 4]
```diff
@@ -175,93 +128,157 @@
...
         NavigationManager.NavigateTo("/browser/" + supervisorId + "/" + endpointId);
     }
 
+    /// <summary>
+    /// BrowseTreeAsync
+    /// </summary>
+    /// <param name="id"></param>
+    /// <param name="parentId"></param>
+    /// <param name="direction"></param>
+    /// <returns></returns>
```
Comment (mregen) [inline samples/src/Microsoft.Azure.IIoT.App/src/Pages/Browser.razor]: ALL of these non informational function comments should be filled in or at least simplified, with a brief summary only...

File: samples/src/Microsoft.Azure.IIoT.App/src/Pages/Browser.razor [hunk 5]
```diff
@@ -175,93 +128,157 @@
...
+                }
+            }
+        }
 
-    private void CloseDrawer()
-    {
-        IsOpened = false;
-        TypeOfAction = ActionType.Nothing;
+        Page = "1";
+        PagedNodeList = NodeList.GetPaged(Int32.Parse(Page), 10);
```
Comment (mregen) [inline samples/src/Microsoft.Azure.IIoT.App/src/Pages/Browser.razor]: >10 [](start = 61, length = 2)

is this pagesize? If yes, have at least a global const to be able to change easily lateron.

File: samples/src/Microsoft.Azure.IIoT.App/src/Services/Publisher.cs [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+                    }
+                } while (!string.IsNullOrEmpty(continuationToken));
+            }
+            catch (Exception e) {
+                // skip this node
+                Trace.TraceError("Cannot get published nodes for endpointId'{0}'", endpointId);
+                var errorMessage = string.Format(e.Message, e.InnerException?.Message ?? "--", e?.StackTrace ?? "--");
+                Trace.TraceError(errorMessage);
+            }
+            pageResult.PageSize = 10;
```
Comment (mregen) [inline samples/src/Microsoft.Azure.IIoT.App/src/Services/Publisher.cs]: > 10; [](start = 33, length = 4)

again the magic number 10 ;-)

**Comment by mregen** [comment]
>             pageResult.PageSize = 10;

Hmm

---
Refers to: samples/src/Microsoft.Azure.IIoT.App/src/Services/Registry.cs:57 in 7057c69. [](commit_id = 7057c695d7f0cbd6f065cebdc6184312d19b0c94, deletion_comment = False)

File: samples/src/Microsoft.Azure.IIoT.App/src/Models/PublishedNode.cs (hunk 1 above)
Comment (dacolgit) [reply samples/src/Microsoft.Azure.IIoT.App/src/Models/PublishedNode.cs]: fixed

File: samples/src/Microsoft.Azure.IIoT.App/src/Pages/Browser.razor (hunk 2 above)
Comment (dacolgit) [reply samples/src/Microsoft.Azure.IIoT.App/src/Pages/Browser.razor]: removed comparison

File: samples/src/Microsoft.Azure.IIoT.App/src/Pages/Browser.razor (hunk 3 above)
Comment (dacolgit) [reply samples/src/Microsoft.Azure.IIoT.App/src/Pages/Browser.razor]: fixed

File: samples/src/Microsoft.Azure.IIoT.App/src/Pages/Browser.razor (hunk 4 above)
Comment (dacolgit) [reply samples/src/Microsoft.Azure.IIoT.App/src/Pages/Browser.razor]: add comments

File: samples/src/Microsoft.Azure.IIoT.App/src/Pages/Browser.razor (hunk 5 above)
Comment (dacolgit) [reply samples/src/Microsoft.Azure.IIoT.App/src/Pages/Browser.razor]: added const value

File: samples/src/Microsoft.Azure.IIoT.App/src/Services/Publisher.cs (hunk 6 above)
Comment (dacolgit) [reply samples/src/Microsoft.Azure.IIoT.App/src/Services/Publisher.cs]: added const value

Comment (mregen) [reply samples/src/Microsoft.Azure.IIoT.App/src/Services/Publisher.cs]: I don't see a difference from the previous commit?

---
In reply to: [356017686](https://github.com/Azure/Industrial-IoT/pull/167#discussion_r356017686) [](ancestors = 356017686)

Comment (mregen) [reply samples/src/Microsoft.Azure.IIoT.App/src/Services/Publisher.cs]: Maybe for the next version you can add an 'IPageConfig' which you define with specific parameters how to display rows/columns

---
In reply to: [355717729](https://github.com/Azure/Industrial-IoT/pull/167#discussion_r355717729) [](ancestors = 355717729)

**Pull request merged**